solr/contrib/ltr/src/test/org/apache/solr/ltr/feature/TestFeatureExtractionFromMultipleSegments.java

```diff
@@ -0,0 +1,116 @@
+                   "func_description", "Some descriptions "+randomString(i%6+3),
+                   "func_description_stemmed", "Some descriptions "+randomString(i%6+3)));
+      
+      assertU(adoc("id", new Integer(i+11).toString(), "global_categoryId", "201",
+                   "comp_description", "apple apple is a company " + randomString(i%6+3), "global_normHits", "0.1",
+                   "func_displayDescription_posBoosted", "New New York New York is New York is Big New York is Big Apple. "+ randomString(i%6+3),
+                   "func_displayDescription_edge", "New York is Big Apple. "+ randomString(i%6+3)));
+      assertU(adoc("id", new Integer(i+12).toString(), "global_categoryId", "201", 
+                   "comp_description", "Big Apple is New York.", "global_normHits", "0.01",
+                   "func_displayDescription_posBoosted", "Big Big Apple Big Apple is Big Apple is New Big Apple is New York. "+ randomString(i%6+3),
+                   "func_displayDescription_edge", "Big Apple is New York. " + randomString(i%6+3)));
+      assertU(adoc("id", new Integer(i+13).toString(), "global_categoryId", "201", 
+                   "comp_description", "New some York is Big. "+ randomString(i%6+3),
+                   "func_displayDescription_posBoosted", "New New some New some York New some York is New some York is Big. "+randomString(i%6+3),
+                   "func_displayDescription_edge", "New some York is Big. "+randomString(i%6+3)));
+      
+      assertU(adoc("id", new Integer(i+14).toString(), "global_categoryId", "201",
+          "comp_description", "apple apple is a company " + randomString(i%6+3), "global_normHits", "0.1",
+          "func_displayDescription_posBoosted", "New New York New York is New York is Big New York is Big Apple. "+ randomString(i%6+3),
+          "func_displayDescription_edge", "New York is Big Apple. "+ randomString(i%6+3)));
+      assertU(adoc("id", new Integer(i+15).toString(), "global_categoryId", "201", 
+          "comp_description", "Big Apple is New York.", "global_normHits", "0.01",
+          "func_displayDescription_posBoosted", "Big Big Apple Big Apple is Big Apple is New Big Apple is New York. "+ randomString(i%6+3),
+          "func_displayDescription_edge", "Big Apple is New York. " + randomString(i%6+3)));
+      assertU(adoc("id", new Integer(i+16).toString(), "global_categoryId", "401", "ppl_normNewsScore", "0.0", 
+                   "global_normHits", "0.0", "ppl_last_name", "OBAMA " + randomString(i%6+3), "ppl_first_name", "barack h"));
+      assertU(adoc("id", new Integer(i+17).toString(), "global_categoryId", "201", "comp_description", "red delicious apple " + randomString(i%6+3), "global_normHits", "0.1"));
+      assertU(adoc("id", new Integer(i+18).toString(), "global_categoryId", "201", "comp_description", "nyc " + randomString(i%6+3), "global_normHits", "0.11"));
+    }
+    
+    assertU(commit());
```

Comment:
Late comment/suggestion here: I think if within the above `for` loop a few commits were added then you could that way achieve the same effect without need for a separate `solrconfig-multiseg.xml` solrconfig variant. Alternatively if the intention and requirement is to not just restrict but to disallow merging then let's configure no-merge-policy in the solrconfig variant.